contrib/gocql/gocql/gocql.go

```diff
@@ -284,3 +303,38 @@ func (tb *Batch) finishSpan(span ddtrace.Span, err error) {
 		span.Finish(tracer.WithError(err))
 	}
 }
+
+type query interface {
+	*gocql.Query | *gocql.Batch
+}
+
+func getClusterContactPoints[T query](q T) ([]string, bool) {
+	defer func() {
+		if r := recover(); r != nil {
+			// ignore panics as we are working with reflection
+		}
+	}()
+	sv := reflect.ValueOf(q).Elem().FieldByName("session")
+	sv = reflect.NewAt(sv.Type(), unsafe.Pointer(sv.UnsafeAddr())).Elem()
+	s, ok := sv.Interface().(*gocql.Session)
+	if !ok {
+		return nil, false
+	}
+	cfg, ok := getClusterConfig(s)
+	if !ok {
+		return nil, false
+	}
+	return cfg.Hosts, true
+}
```

Comment:
Isn't this the same reason why we didn't merge https://github.com/DataDog/dd-trace-go/pull/1765 ?

Reply:
@ahmed-mez yes! don't bother about this PR for now, I'm planning to setup a discussion offline about considering the usage of reflection in contribs (we can maybe use this PR as an example use-case). I'll move back this PR to draft for now.